controllers/nestedcluster_controller.go

```diff
@@ -48,10 +59,51 @@ type NestedClusterReconciler struct {
 }
 
 // SetupWithManager sets up the controller with the Manager.
-func (r *NestedClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+func (r *NestedClusterReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	clusterToInfraFn := util.ClusterToInfrastructureMapFunc(infrav1.GroupVersion.WithKind("NestedCluster"))
+	log := ctrl.LoggerFrom(ctx)
+
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&infrav1.NestedCluster{}).
-		Owns(&controlplanev1.NestedControlPlane{}).
+		WithOptions(options).
+		For(&infrav1.NestedCluster{},
+			builder.WithPredicates(
+				predicate.Funcs{
+					// Avoid reconciling if the event triggering the reconciliation is related to incremental status updates
```

Comment:
Were you seeing this happening frequently? I haven't experienced this before but I also could have just missed it.

Reply:
this is copied from CAPO repo code, https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/748
I admit I didn't have such as well..